### PR TITLE
V4.3.x

### PR DESCRIPTION
**Pull request by JamieVangeysel**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -7,15 +7,22 @@ export interface IFastifyConfig {
...
+    additionalSecurityHeaders?: boolean
   },
   elastic?: {
     node?: string,
     index: string,
     auth: {
       username: string,
       password: string
-    }
+    },
```
Comment (OhSoooLucky) [inline index.d.ts]: ?

File: index.js [hunk 2]
```diff
@@ -52,8 +71,11 @@ module.exports = async function (appConfig) {
   if (config.logger == null)
     config.logger = true
   else if (config.logger !== true && appConfig.elastic)
-    // If elastic is configured, use pine with pine-elasticsearch
-    config.logger = setupLogging(appConfig.elastic, config.logger, appConfig.serviceName)
+    // If elastic is configured, use pino with pino-elasticsearch
+    config.logger = setupElasticLogging(appConfig.elastic, config.logger, appConfig.serviceName)
+  else if (config.logger !== true && appConfig.logtail)
```
Comment (OhSoooLucky) [inline index.js]: Double check for config.logger !== true, nest instead

**Review by OhSoooLucky — CHANGES_REQUESTED** [review]

File: index.js (hunk 2 above)
Comment (JamieVangeysel) [reply index.js]: nesting increases code complexity by 2, wontfix

Comment (JamieVangeysel) [reply index.js]: will create an separate function setupLogging

**Review by OhSoooLucky — APPROVED** [review]
LTM... G

**Pull request merged**